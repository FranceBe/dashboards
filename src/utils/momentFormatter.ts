import moment from 'moment'

export const momentFormatter = (date: string): string =>
  moment(date).format('MMMM Do YYYY, h:mm:ss a')
