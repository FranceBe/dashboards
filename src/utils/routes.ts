import GlobalDashboard from 'containers/GlobalDashboard'
import { Routable } from 'types/route'

export const routes: Routable[] = [
  {
    component: GlobalDashboard,
    exact: true,
    path: '/',
  },
]
