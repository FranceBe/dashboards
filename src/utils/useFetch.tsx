// Custom hook to call an API using fetch and cache
// Cache handling could be improved if we want to get
// real-time information or if we want to get a "refresh" action
import { useEffect, useState } from 'react'

// Creating cache to avoid recalling api if data has already been fetched
const cache: Record<string, any> = {}

export const useFetch = <T,>(url: string): { data?: T | T[]; status: string } => {
  const [status, setStatus] = useState('idle')
  const [data, setData] = useState()

  useEffect(() => {
    let shouldUpdateState = true
    if (!url) return
    setStatus('fetching')
    if (cache[url]) {
      // if cache is filled, do not call fetch
      // use cache to fill the data state
      const data = cache[url]
      setData(data)
      setStatus('fetched')
    } else {
      // if cache is not filled
      // call fetch to get the desired data
      fetch(url)
        .then((res) => res.json())
        .then((data) => {
          if (shouldUpdateState) {
            // In this scenario the backend api 'quizz-back'
            // respond with either a key "results" and an array (/devices)
            // or with an object with no "result" key (/device/id)
            // In order to get data correctly, we check if
            // the key "results" exists before saving the data we want
            cache[url] = data.results ? data.results : data
            setData(data.results ? data.results : data)
          }
        })
        .catch((err) => {
          throw Error(err)
        })
      setStatus('fetched')
    }
    return () => {
      shouldUpdateState = false
    }
  }, [url])
  return { data, status }
}
