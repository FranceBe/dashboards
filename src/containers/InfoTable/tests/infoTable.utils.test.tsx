import { columns } from 'containers/InfoTable/infoTable.utils'

describe('InfoTable Utils', () => {
  describe('columns', () => {
    it('should match snapshot', () => {
      expect(columns).toMatchSnapshot()
    })
  })
})
