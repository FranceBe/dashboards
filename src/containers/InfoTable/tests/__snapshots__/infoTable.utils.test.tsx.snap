// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoTable Utils columns should match snapshot 1`] = `
Array [
  Object {
    "name": "serial_number",
    "selector": [Function],
    "sortFunction": [Function],
    "sortable": true,
  },
  Object {
    "name": "mac_wifi",
    "selector": [Function],
    "sortFunction": [Function],
    "sortable": true,
  },
  Object {
    "name": "sim_id",
    "selector": [Function],
    "sortable": true,
  },
  Object {
    "name": "last_seen_at",
    "selector": [Function],
    "sortable": true,
  },
  Object {
    "grow": 0,
    "name": "connection_type",
    "selector": [Function],
    "sortFunction": [Function],
    "sortable": true,
  },
  Object {
    "grow": 0,
    "name": "status",
    "selector": [Function],
    "sortFunction": [Function],
    "sortable": true,
  },
]
`;
